packages/test-helper/src/path.ts

```diff
@@ -26,7 +26,7 @@ export function compilePathMatcherRegExp(match: string | RegExp) {
     return match;
   }
   const escaped = _.escapeRegExp(match);
-  return new RegExp(`(?<=\\W|^)${escaped}(?=\\W|$)`);
+  return new RegExp(`(?<=\\W|^)${escaped}(?=\\W|$)`, 'g');
```

Comment:
Note this change. 

Without the global flag, the RegExp will only take effect on the first matchs. This could be mistaken when there more than one possible matches in the snapshot.

As well as the change in `packages/test-helper/src/pathSerializer.ts` file.

Reply:
LGTM 👍🏻

Reply:
Found that the Windows UT is failed:

![image](https://github.com/web-infra-dev/rsbuild/assets/7237365/3f1fb8de-2f5c-40f6-a0f4-c7dca155d553)

https://github.com/web-infra-dev/rsbuild/actions/runs/6692882842/job/18182948873?pr=260